### PR TITLE
Make the BugExtractor apply rollback with multiple processes

### PR DESCRIPTION
**Pull request by marco-c**

### CODE REVIEW

File: bugbug/bug_features.py [hunk 1]
```diff
@@ -808,10 +807,21 @@ def apply_transform(bug):
...
+            if self.rollback:
+                with Pool() as p:
+                    return pd.DataFrame(
+                        apply_transform(bug)
+                        for bug in p.imap(
+                            partial(bug_snapshot.rollback, when=self.rollback_when),
+                            bugs_iter,
+                            chunksize=1024,
+                        )
+                    )
```
Comment (suhaibmujahid) [inline bugbug/bug_features.py:820]: This is kind of deeply nested. We could extract it in a function (e.g., `apply_rollback()`) that returns a generator of rolled-back bugs. With that, if need to rollback, we just replace the bugs iterator:
```py
if self.rollback:
    bugs_iter = apply_rollback(bugs_iter)
return pd.DataFrame(apply_transform(bug) for bug in bugs_iter)
```

**Review by suhaibmujahid — DISMISSED** [review]
LGTM! I just have a readability suggestion.

File: bugbug/bug_features.py (hunk 1 above)
Comment (marco-c) [reply bugbug/bug_features.py]: I've improved this a bit in 71e7552.

**Review by suhaibmujahid — DISMISSED** [review]

File: bugbug/bug_features.py [hunk 2]
```diff
@@ -808,10 +807,19 @@ def apply_transform(bug):
...
+            if self.rollback:
+                with Pool() as p:
+                    bugs_iter = p.imap(
+                        partial(bug_snapshot.rollback, when=self.rollback_when),
+                        bugs_iter,
+                        chunksize=1024,
+                    )
+                    return pd.DataFrame(apply_transform(bug) for bug in bugs_iter)
+            else:
+                return pd.DataFrame(apply_transform(bug) for bug in bugs_iter)
```
[the remark this answers is not in the document]
Comment (suhaibmujahid) [reply bugbug/bug_features.py:820]: I see. I guess the approach in https://github.com/mozilla/bugbug/pull/3780#discussion_r1379489127 will not impact the context manager if we `yield` the bugs from within the `apply_rollback()` function.

Anyway, I'm OK with this.

[the remark this answers is not in the document]
Comment (marco-c) [reply bugbug/bug_features.py]: Good point, done in d571a77.

**Pull request merged**